### PR TITLE
Remove jQuery from plugin [CU-1upan8]

### PR DESCRIPTION
**Pull request by debbieellis7**
Resolved: #54 

Remove jQuery as dependency.

### CODE REVIEW

File: assets/js/main.js [hunk 1]
```diff
@@ -1,51 +1,69 @@
...
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
 /**
  * The JS entry point for Webpack
  *
  * @since 0.1
  */
-$(document).bind('gform_post_render', function (e, formId) {
+jQuery(document).bind('gform_post_render', function (e, formId) {
```
Comment (jakejackson1) [inline assets/js/main.js:35]: This is just capturing a custom event. You should be able to `addEventListener` on `gform_post_render`

File: assets/js/main.js [hunk 2]
```diff
@@ -1,51 +1,69 @@
...
   }
 
-  let $form = $('#gform_' + formId)
+  let form = document.getElementById('gform_' + formId)
 
   /* Try match a slightly different mark-up */
-  if ($form.length == 0) {
-    $form = $('#gform_wrapper_' + formId).closest('form')
+  if (form.length === 0) {
+    console.log('form')
```
Comment (jakejackson1) [inline assets/js/main.js:45]: remove this

File: assets/js/main.js [hunk 3]
```diff
@@ -1,51 +1,69 @@
...
-  $form.find('.gpdf-previewer-wrapper').each(function () {
+  const wrapper = form.getElementsByClassName('gpdf-previewer-wrapper')
+  const elem = [].slice.call(wrapper)
 
-    let fieldId = parseInt($(this).data('field-id'))
-    let pdfId = $(this).data('pdf-id')
-    let previewerHeight = parseInt($(this).data('previewer-height'))
-    let download = (typeof $(this).data('download') !== 'undefined') ? parseInt($(this).data('download')) : 0;
+  elem.map(item => {
+    let fieldId = parseInt(item.getAttribute('data-field-id'))
```
Comment (jakejackson1) [inline assets/js/main.js:54]: You don't need to run this through `parseInt` as it's only used in `string` context

File: assets/js/main.js [hunk 4]
```diff
@@ -1,51 +1,69 @@
...
+    let pdfId = item.getAttribute('data-pdf-id')
+    let previewerHeight = parseInt(item.getAttribute('data-previewer-height'))
+    let download = item.getAttribute('data-download') != null ? parseInt(item.getAttribute('data-download')) : 0
```
Comment (jakejackson1) [inline assets/js/main.js:57]: You don't need to run this through `parseInt` as it's only used in `string` context

File: assets/js/main.js [hunk 5]
```diff
@@ -1,51 +1,69 @@
...
 
     /* Continue to next matched element if no PDF ID exists */
     if (pdfId == 0) {
+      console.log('form 2')
```
Comment (jakejackson1) [inline assets/js/main.js:61]: remove this.

File: assets/js/main.js [hunk 6]
```diff
@@ -56,12 +74,13 @@ $(document).bind('gform_post_render', function (e, formId) {
     })
 
     let previewer = new Generator({
-      form: $form,
-      container: $(this),
+      form: form,
+      formId: formId,
+      container: wrapper[0],
```
Comment (jakejackson1) [inline assets/js/main.js:79]: Will multiple Previewer work on the same page using this method?

File: assets/js/main.js (hunk 1 above)
Comment (debbieellis7) [reply assets/js/main.js:35]: Can't @jakejackson1 . jQuery provides an event layer over vanilla JS. That means that vanilla JS cannot talk to that added layer. jQuery can catch vanilla JS events, but vanilla JS cannot catch jQuery added events.

Comment (jakejackson1) [reply assets/js/main.js:35]: Ah, good to know!

**Comment by gravitypdf-travis** [comment]
Task linked: [CU-1upan8 Strip jQuery](https://app.clickup.com/t/1upan8)

**Comment by jakejackson1** [comment]
We will need to prevent the Previewer being generated if the DOM element is currently hidden. For example, using the "Page Break" field, if you add a Previewer field on Page 2 of the form and then load Page 1 you can see that the Previewer is automatically loaded on page 2.

**Comment by gravitypdf-travis** [comment]
-- Please reply above this line --


Hey Gravitypdf/gravity-Pdf-Previewer,

Thanks for reaching out to the Gravity PDF team.

This is an automated message to let you know we've received your email
and one of our team will get back to you in the next 24-48 hours.

Please be aware, our support hours are 9:00am-5:00pm Monday to Friday,
Sydney Australia time [1] (public holidays excluded).

Speak soon,

– Gravity PDF Team

Links:
------
[1] https://www.timeanddate.com/worldclock/australia/sydney

**Comment by jakejackson1** [comment]
Instead of waiting until the Previewer DOM element is 100% in the viewport to automatically reload, for a better experience we should handle it as soon as a part of the DOM element is in the viewport.

**Comment by jakejackson1** [comment]
Let's take a leaf out of the `takeLatest` saga book for handling events. For example, if the user makes a change to a field and then scrolls down the Previewer will begin generating a new preview. If they then modify another field and scroll down they will have to wait for the original Preview request to complete and then generate a new request to see the latest data.

I'd like to automatically cancel that first request and load a new one when this occurs.

**Comment by jakejackson1** [comment]
@debbieellis7 here's a form export you can use to easily test the auto-refresh functionality and the multi-page load problem mentioned above. Just remove the `.txt` from the end of the export file.
[gravityforms-export-2020-06-02.json.txt](https://github.com/GravityPDF/gravity-pdf-previewer/files/4714301/gravityforms-export-2020-06-02.json.txt)

**Comment by jakejackson1** [comment]
image
When PDF error occurs the `<br>` isn't processed into HTML. If you click refresh, the error doesn't get removed before the new error is displayed.

**Comment by jakejackson1** [comment]
image

Previewer no longer working with the [Gravity Wiz Live Preview perk](https://gravitywiz.com/documentation/gravity-forms-live-preview/).

**Comment by jakejackson1** [comment]
Because many Gravity Forms plugins use jQuery's event handling, we'll need to use jQuery's `change` event on the `form` to correctly listen for all form changes. As you mentioned, using the native event listeners misses any jQuery events.

File: assets/js/Previewer/Generator.js [hunk 7]
```diff
@@ -1,8 +1,9 @@
-import $ from 'jquery'
 import debounce from 'debounce'
-import inViewport from 'in-viewport'
+import 'abortcontroller-polyfill/dist/abortcontroller-polyfill-only'
```
Comment (jakejackson1) [inline assets/js/Previewer/Generator.js:2]: Note to self: This is needed to fix an error in the Unit test. I'd like to see if doing ` const AbortController = window.AbortController;` and then running `new AbortController()` resolves this issue without the Polyfil.

**Pull request merged**